src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1369,9 +1369,18 @@ private static bool IsCompilerAnalyzer(DiagnosticAnalyzer analyzer)
 
         public void Dispose()
         {
-            this.CompilationEventQueue?.TryComplete();
-            this.DiagnosticQueue?.TryComplete();
-            _queueRegistration.Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
```

Comment:
I do not believe there is such a requirement. There is a recommendation about implementing IDisposable for classes that have finalizers/destructors, but that requires both implementations to be coordinated with each other. Simply adding one part doesn't feel useful to me (even in a separate PR), either we should complete the pattern, or remove the new method.